src/MongoDB.Analyzer/Core/Builders/AnalysisCodeGenerator.cs

```diff
@@ -69,16 +69,16 @@ public static CompilationResult Compile(MongoAnalysisContext context, Expression
     private static SyntaxTree GenerateMqlGeneratorSyntaxTree(ExpressionsAnalysis builderExpressionAnalysis)
     {
         var testCodeBuilder = new BuildersMqlGeneratorTemplateBuilder(s_mqlGeneratorSyntaxElements);
+        var generatedMqlMethodDeclarations = new List<MethodDeclarationSyntax>();
 
         foreach (var builderContext in builderExpressionAnalysis.AnalysisNodeContexts)
         {
-            var analysisNode = builderContext.Node;
-
-            builderContext.EvaluationMethodName = testCodeBuilder.AddBuildersExpression(
-                analysisNode.ArgumentTypeName,
-                analysisNode.RewrittenExpression);
+            var (generatedMqlMethodName, generatedMqlMethodDeclaration) = testCodeBuilder.GenerateMqlGeneratorMethod(builderContext);
+            builderContext.EvaluationMethodName = generatedMqlMethodName;
+            generatedMqlMethodDeclarations.Add(generatedMqlMethodDeclaration);
```

Comment:
It's better to pass down to `GenerateMqlGeneratorMethod` the minimal data it needs, which is `ArgumentTypeName` and `RewrittenExpression`. It should not be aware of `ExpressionAnalysisNode` at this point.

Reply:
Done